File: scripts/dataframe_to_GRanges.R
source("scripts/helpers.R")
library(GenomicRanges)
library(data.table)

f = snakemake@input[[1]]
df = get_df(f)

start.time <- Sys.time()

gr = GRanges(seqnames = df$Chromosome,
             ranges = IRanges(start = df$Start, end = df$End), strand = df$Strand)

end.time <- Sys.time()

time.taken <- end.time - start.time

time.taken <- as.numeric(time.taken, units="secs")

write(time.taken, snakemake@output[[1]])

File: scripts/dataframe_to_GRanges.py
from helpers import read_file, init_ray; init_ray(snakemake.wildcards)
from pyranges import PyRanges

from time import time
import datetime

f = snakemake.input[0]
chip = read_file(f)

start = time()
granges = PyRanges(chip, copy_df=False)
end = time()
total = end - start

total_dt = datetime.datetime.fromtimestamp(total)

minutes_seconds = total_dt.strftime('%-M.%-S.%f')

open(snakemake.output[0], "w+").write(minutes_seconds)
